skimage/util/points_to_labels.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+__all__ = ["label_points"]
+
+
+def label_points(coords, image):
+    """Assign unique integer labels to coordinates on an image mask
+
+    Parameters
+    ----------
+    coords: ndarray
+        An array of 2D coordinates with shape (n, 2)
+    image: ndarray
+        Image for which the mask is created
+
+    Returns
+    -------
+    label_mask: ndarray
+        A 2D mask of zeroes containing unique integer labels at the `coords`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage.util.points_to_labels import label_points
+    >>> coords = np.array([[0, 1], [2, 2]])
+    >>> image = np.zeros((5, 5), dtype=np.uint32)
+    >>> mask = label_points(coords, image)
+    >>> mask
+    array([[0, 1, 0, 0, 0],
+           [0, 0, 0, 0, 0],
+           [0, 0, 2, 0, 0],
+           [0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0]], dtype=uint64)
+    """
+    if not isinstance(coords, np.ndarray):
+        raise TypeError("'coords' must be an ndarray")
+
+    if coords.shape[-1] != 2 or coords.ndim != 2:
+        raise ValueError("'coords' must be of shape (n, 2)")
+
+    h, w = image.shape[:2]
```

Comment:
If we want to support use of this function on multichannel and n-dimensional images, then we should probably add a `multichannel` keyword argument (defaulting to False). This way we don't have to try and guess based on shape if the image is multichannel. If this `multichannel` argument is present then, we can get the shape of the spatial dimensions using something like:
```Python
shape  = image.shape[:-1] if multichannel else image.shape
```

This can be moved above the shape check above so that you can check `if coords.shape[-1] != len(shape)` instead of assuming 2 dimensions.

Reply:
Huh? We shouldn't use multichannel here, all we need is the output shape. ie don't pass in an image as an argument, only the shape.